app/src/main/java/org/simple/clinic/overdue/AppointmentRepository.kt

```diff
@@ -2,13 +2,40 @@ package org.simple.clinic.overdue
 
 import io.reactivex.Completable
 import io.reactivex.Single
+import org.simple.clinic.facility.FacilityRepository
 import org.simple.clinic.patient.SyncStatus
 import org.simple.clinic.patient.canBeOverriddenByServerCopy
 import org.simple.clinic.sync.SynceableRepository
+import org.simple.clinic.user.UserSession
+import org.threeten.bp.Instant
+import org.threeten.bp.LocalDate
 import java.util.UUID
 import javax.inject.Inject
 
-class AppointmentRepository @Inject constructor(val dao: Appointment.RoomDao) : SynceableRepository<Appointment, AppointmentPayload> {
+class AppointmentRepository @Inject constructor(
+    val dao: Appointment.RoomDao,
+    val userSession: UserSession,
+    val facilityRepository: FacilityRepository
+) : SynceableRepository<Appointment, AppointmentPayload> {
+
+  fun schedule(patientId: UUID, appointmentDate: LocalDate): Completable {
+    return facilityRepository
+        .currentFacility(userSession)
+        .take(1)
+        .flatMapCompletable { facility ->
```

Comment:
Can you move the construction of the `Appointment` instance into a `map()` operator?

Reply:
Yea okay